### PR TITLE
gave the arch wiki a dark mode toggle

### PR DESCRIPTION
**Pull request by killertofus**
any suggestions are welcome

### CODE REVIEW

**Comment by pierres** [comment]
Instead of copying the extension into our project could you reference it with a git submodule? This makes maintenance a lot easier. See the CheckUser extension for example.

File: LocalSettings.archlinux.org.php [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+# temporary disable registration; reverted 2016-6-26 bluewind
+#$wgGroupPermissions['*']['createaccount'] = false;
+
+# $wgShowSQLErrors = true;
+# $wgReadOnly = 'Database migration in progress. We`ll be back in a few minutes.';
+# made the wiki have a dark mode
+#wfLoadExtension( 'DarkMode' );
+$wgDarkModeTogglePosition
+#sidebar
+
```
Comment (nl6720) [inline LocalSettings.archlinux.org.php:378]: The `LocalSettings.archlinux.org.php` file wasn't [removed](https://github.com/archlinux/archwiki/commit/058c6cbd5b0eb24934afab5b34147a060920922b) without a reason. Even if you add it back, nothing will use it.

ArchWiki's `LocalSettings.php` is in the infrastructure repo: https://gitlab.archlinux.org/archlinux/infrastructure/-/blob/master/roles/archwiki/templates/LocalSettings.php.j2.

**Comment by killertofus** [comment]
@nl6720 @pierres should be fixed

**Comment by lahwaacz** [comment]
There are 2 submodules now: in `mediawiki-extensions-DarkMode` and `extensions/mediawiki-extensions-DarkMode`.

And `LocalSettings.php.j2` does not belong in this repo - it needs to be maintained in the [infrastructure repo](https://gitlab.archlinux.org/archlinux/infrastructure/).

Also please squash your commits in the end.

File: .gitmodules [hunk 2]
```diff
@@ -26,3 +26,6 @@
 	path = extensions/FlarumAuth
 	url = https://github.com/archlinux-de/mediawiki-extensions-FlarumAuth.git
 	branch = REL1_38
+[submodule "extensions/mediawiki-extensions-DarkMode"]
+	path = extensions/mediawiki-extensions-DarkMode
+	url = https://github.com/wikimedia/mediawiki-extensions-DarkMode
```
Comment (nl6720) [inline .gitmodules:31]: The git submodule branch needs to match the used mediawiki release.

```suggestion
[submodule "extensions/DarkMode"]
	path = extensions/DarkMode
	url = https://github.com/wikimedia/mediawiki-extensions-DarkMode.git
	branch = REL1_38
```


The submodule currently should be at commit `fe470966d18fa4749f9b92097052fcd031e3cad6`.

**Review by nl6720 — CHANGES_REQUESTED** [review]

**Review by nl6720 — CHANGES_REQUESTED** [review]
The submodule is still not properly added.
It should be in `extensions/DarkMode` not `extensions/mediawiki-extensions-DarkMode` and at the latest commit of the `REL1_38` branch.

**Review by killertofus — COMMENTED** [review]
@nl6720 it is

**Comment by lahwaacz** [comment]
@killertofus It's not only about the `.gitmodules` file, the diff shows:
```
extensions/mediawiki-extensions-DarkMode
 Submodule mediawiki-extensions-DarkMode added at b36d0f
```

**Pull request closed without merging**